### PR TITLE
docs: new entry about webchat connection and disconnection

### PR DESCRIPTION
**Pull request by annehubtype**
Please review:

- New menu entry added under **Webchat > Connectivity**
- **sidebars.json** updated for doc tool purpose.
- **webchat-api** updated with `Botonic.isOnline()`

## Testing

The pull request...

- has unit tests
- has integration tests
- doesn't need tests because... **[provide a description]**

### CODE REVIEW

File: docs/docs/webchat/webchat-api.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ These methods are:
 - `Botonic.openCoverComponent()`: Opens an authentication window.
 - `Botonic.closeCoverComponent()`: Closes the authentication window.
 - `Botonic.toggleCoverComponent()`: Opens or closes the authentication window depending on the current state.
+- `Botonic.isOnline(boolean)`: Sets the webchat connection status.
```
Comment (vanbasten17) [inline docs/docs/webchat/webchat-api.md:23]: `Gets the webchat connection status to know whether is connected (`true`) or not (`false`)`.

File: docs/docs/webchat/webchat-connectivity.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+id: webchat-connectivity
+title: Connectivity
+---
+
+Sometimes, the connection can be lost between the server and the chatbot. To handle this situation, you can set specific parameters to quickly detect the connection issue.
```
Comment (vanbasten17) [inline docs/docs/webchat/webchat-connectivity.md:6]: To handle this situation, you can set specific parameters to tune how to detect the connection issue.

File: docs/docs/webchat/webchat-connectivity.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+For a better user experience, you can also warn the user by configuring a customized error message to be displayed within the webchat interface.
+
+To do so:
```
Comment (vanbasten17) [inline docs/docs/webchat/webchat-connectivity.md:10]: To do so, you can configure the server options to tune timings for resolving Webchat's connection.

File: docs/docs/webchat/webchat-connectivity.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+1. Set the `Botonic.isOnline()` method. Values are `true` or `false` depending on your server configuration.
```
Comment (vanbasten17) [inline docs/docs/webchat/webchat-connectivity.md:12]: Change `To do so` above with the mentioned suggestion or similar and remove these two points. The first one is not accurated.

File: docs/docs/webchat/webchat-connectivity.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+```javascript
+export const webchat = {
+  server: {
+    activityTimeout: 30 * 1000,
+    pongTimeout: 10 * 1000,
+    errorMessage: 'Connection issue',
+  },
+}
+```
+You can use a function if you prefer. For example, if you want to display the warning message in various languages:
```
Comment (vanbasten17) [inline docs/docs/webchat/webchat-connectivity.md:39]: if you want to display the warning message depending on the language:

File: docs/docs/webchat/webchat-connectivity.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    pongTimeout: 20 * 1000,
+  },
+}
+```
+
+- `activityTimeout`: Time to wait (in ms) before pinging the server to know if the connection is still working. Default value: 20 * 1000.
+
+- `pongTimeout`: After pinging the server, time to wait (in ms) before assuming the connection is lost and closing it. Default value: 5 * 1000.
+
+3. Add the `errorMessage` variable to change the warning message when a connection issue occurs.
```
Comment (vanbasten17) [inline docs/docs/webchat/webchat-connectivity.md:28]: Remove this point and add a description like:
`To define the message to be displayed when webchat experiences connecitivity issues you can use `errorMessage``

**Review by vanbasten17 — CHANGES_REQUESTED** [review]

**Review by vanbasten17 — APPROVED** [review]

File: docs/docs/webchat/webchat-api.md [hunk 7]
```diff
@@ -20,6 +20,8 @@ These methods are:
 - `Botonic.openCoverComponent()`: Opens an authentication window.
 - `Botonic.closeCoverComponent()`: Closes the authentication window.
 - `Botonic.toggleCoverComponent()`: Opens or closes the authentication window depending on the current state.
+- `Botonic.isOnline(boolean)`: Gets the webchat connection status to know whether it is connected (true) or not (false).
```
Comment (vanbasten17) [inline docs/docs/webchat/webchat-api.md:23]: `Botonic.isOnline(boolean)` --> ``Botonic.isOnline()``

File: docs/docs/webchat/webchat-connectivity.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
+---
+id: webchat-connectivity
+title: Connectivity
+---
+
+Sometimes, the connection can be lost between the server and the chatbot. To handle this situation, you can set specific parameters to detect the connection issue.
```
Comment (ericmarcos) [inline docs/docs/webchat/webchat-connectivity.md:6]: "between the server and the **chatbot**" -> "between the server and the **webchat**"

File: docs/docs/webchat/webchat-connectivity.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+```javascript
+export const webchat = {
+  server: {
+    activityTimeout: 50 * 1000,
+    pongTimeout: 20 * 1000,
+  },
+}
+```
+
+- `activityTimeout`: Time to wait (in ms) before pinging the server to know if the connection is still working. Default value: 20 * 1000.
```
Comment (ericmarcos) [inline docs/docs/webchat/webchat-connectivity.md:23]: Default value 20 seconds (20*1000)

File: docs/docs/webchat/webchat-connectivity.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+- `pongTimeout`: After pinging the server, time to wait (in ms) before assuming the connection is lost and closing it. Default value: 5 * 1000.
```
Comment (ericmarcos) [inline docs/docs/webchat/webchat-connectivity.md:25]: Default value 5 seconds (5*1000)

File: docs/docs/webchat/webchat-connectivity.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+```
+You can use a function if you prefer. For example, if you want to display a warning message depending on the language:
+
+```javascript
+export const webchat = {
+  server: {
+    activityTimeout: 20 * 1000,
+    pongTimeout: 5 * 1000,
+    errorMessage: () => {
+       if(locale === 'es') return 'Problemas de conexión'
```
Comment (ericmarcos) [inline docs/docs/webchat/webchat-connectivity.md:48]: This variable is not defined anywhere, I don't like when documentation examples give me incomplete code

**Review by ericmarcos — CHANGES_REQUESTED** [review]
As a developer I'd like to see a more general description of what the framework supports in terms of connectivity, how does it do it, and finally what options do I have to personalise this behaviour. For example:

Botonic comes with offline support, meaning that messages sent / received while the user loses connection are not lost. Instead, they're stored in a queue and are re-sent once connection is recovered. Botonic detects if the user is online/offline by sending ping-pong messages through the websocket connection with the server. By default, Botonic sends a "ping" every 20 seconds and waits for a "pong" response for 5 seconds, if this response is not received then a "Connection issues" indicator is shown to the user (screenshot).
You can customize this behaviour...

**Review by ericmarcos — APPROVED** [review]

**Pull request merged**